### DISCUSSION TITLE
Table stuck in Load

### DISCUSSION THREAD

**Discussion by ElisameAraujo** [Power-Components/livewire-powergrid#1131] [Q&A]
I'm trying to add a powergrid table but even though it renders data from the database it gets stuck in infinite load. I cannot filter, search or any other action.

I've tested using Eloquent and Model, but it gets stuck in this state.

Another thing, how could I change the Primary Key to the column name I use? Because by default powergrid looks for the "id" column, but the one I use is called "artistaID".

![image](https://github.com/Power-Components/livewire-powergrid/assets/40779799/7c627c4d-892f-41f9-b4e3-154586a58287)

**Comment by abhaypithadiya** [comment]
Hey! Could you check if there are any errors in your console for the table load issue and to change your primary key to a custom column you can add the following:
```
final class RestaurantsTable extends PowerGridComponent
{
    use ActionButton;

    public string $primaryKey = 'artistaID';
    
     //...
```
More about it in the documentation : https://v4.livewire-powergrid.com/table/datasource.html#custom-primary-key

**Comment by ElisameAraujo** [comment]
No errors are shown, neither on the browser console, or within the VSCode console.

It's strange, because using DataTables, the table loads the data normally and without problems.

I followed this [tutorial](https://www.youtube.com/watch?v=Qj0GLZJzDLY), but it is precisely when the table is rendered that it enters loop.

**Reply by abhaypithadiya** [reply]
Which Powergrid Version are you using?

**Reply by ElisameAraujo** [reply]
the latest version with livewire v2.12.6

**Comment by luanfreitasdev** [comment]
Hello!. I fixed this problem in this PR and I also fixed it in version 5.x.

https://github.com/Power-Components/livewire-powergrid/pull/1134
